### PR TITLE
feat(): I/O Registry

### PR DESCRIPTION
**Pull request by ShaMan123**
```ts

import { registry } from 'fabric';

class Mip {
    type = 'mip'
    static async fromObject(d) {
        return new Mip();
    }
    static fromElement(a, b) {
        return new Mip();
    }
}

registry.registerClass(Mip);

```

```ts
const mipInstance = registry.fromJSON({ type: 'mip', ...})
```

### CODE REVIEW

**Comment by ShaMan123** [comment]
I think we can drop `type` in favor of `klass.name` and do a switch for SVG

**Comment by ShaMan123** [comment]
The other issue is how to customize/subclass the registry

File: src/Registry.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+        return this.registry.get(data.type)?.fromJSON(data);
+    }
+
+    resolveSVGType(el: SVGElement) {
+        return el.tagName.replace('svg:', '');
+    }
+
+    fromSVG<T extends SVGElement, S>(el: T, options: S) {
+        return this.registry.get(this.resolveSVGType(el))?.fromSVG(el, options);
+    }
```
Comment (asturur) [inline src/Registry.ts:46]: Why the registry can't just answer to the question:
what can i enlive for object.type 'rect'
what can i enlive fro svg element 'rect'
?

Why does it have to call the methods too?

Comment (asturur) [reply src/Registry.ts:46]: something like backward:

```js
this.register(klass, {
            type: klass.type,
            svgElement: 'rect'
        });
```

I think registering the methods unless has extra benefits i don't get withut an explanation, is just an extra thing a developer has to parse/learn and follow when is debugging code

Comment (ShaMan123) [reply src/Registry.ts:46]: I thought of hybrid things that don't conform to how objects/classes are defined in fabric

Comment (ShaMan123) [reply src/Registry.ts:46]: This is good

> something like backward:
> 
> ```js
> this.register(klass, {
>             type: klass.type,
>             svgElement: 'rect'
>         });
> ```
>

Comment (ShaMan123) [reply src/Registry.ts:46]: Is there a use case that an svg tag maybe parsed by fabric to 2+ different objects?
Like a group containing elements, mask etc.
That is why

Comment (asturur) [reply src/Registry.ts:46]: I don't know seems an unnecessary new thing.
A fixed map is bad becuase it makes you import everything.
A map on the go is fine, while you import you register, when you create your class you can override an import with your.

I can see someone wanting to parse and svg text with textbox or itext more than text for example, or a special path of some sort ( an editable path ).

I understand that the relationship between className and type is very rudimental and should be removed, but i really don't want anything that is not as simple as registrable map.

Given a type you give me the class, given the svg tag you give me the class.
Anything above this should have tangible benefits for current use cases.

Comment (ShaMan123) [reply src/Registry.ts:46]: You stated the benefits.
Just to make sure I understand:
"Given a type you give me the function (toJSON in most cases), given the svg tag you give me the function (toSVG)"
falls under your definition or not?

**Comment by ShaMan123** [comment]
closed in favor of #8282

**Pull request closed without merging**